pkg/resources/data.go

```diff
@@ -688,12 +691,23 @@ func GetCSIMigrationFeatureGates(cluster *kubermaticv1.Cluster) []string {
 		// The following feature gates are always enabled when the
 		// 'externalCloudProvider' feature is activated.
 		if cluster.Spec.Features[kubermaticv1.ClusterFeatureExternalCloudProvider] {
-			featureFlags = append(featureFlags, "CSIMigration=true", "CSIMigrationOpenStack=true", "ExpandCSIVolumes=true")
+			featureFlags = append(featureFlags, "CSIMigration=true", "ExpandCSIVolumes=true")
+		}
+		if cluster.Spec.Cloud.Openstack != nil {
+			featureFlags = append(featureFlags, "CSIMigrationOpenStack=true")
+		}
+		if cluster.Spec.Cloud.VSphere != nil {
+			featureFlags = append(featureFlags, "CSIMigrationvSphere=true")
 		}
```

Comment:
Do we also need to enable the CSI migration here?

https://github.com/kubermatic/kubermatic/blob/5182eb92494a74f20d7e3941179bd63df99d537d/addons/csi/vsphere/csiController.yaml#L193

If yes, please take care of that in a follow-up.